src/main/java/step3/Statistics.java

```diff
@@ -0,0 +1,45 @@
+package step3;
+
+import step3.util.MathUtils;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Statistics {
+    private static final Map<WinningStatistics, Integer> statistics = new HashMap<>();
+    private static final int LOTTO_PRICE = 1000;
+
+    private Statistics() {
+        for (WinningStatistics value : WinningStatistics.values()) {
+            statistics.put(value, 0);
+        }
+    }
+
+    public Statistics(List<LottoResult> winningCountList) {
+        this();
+
+        winningCountList.stream()
+                .filter(lottoResult -> WinningStatistics.checkMatchedCount(lottoResult.getWinningCount()))
+                .forEach(lottoResult -> {
+                    WinningStatistics key = WinningStatistics.valueOfWinningCount(lottoResult.getWinningCount(), lottoResult.hasBonus());
+                    int value = statistics.get(key);
+                    statistics.put(key, ++value);
+                });
+    }
+
+    public Map<WinningStatistics, Integer> getStatistics() {
+        return statistics;
+    }
+
+    public double calculateYield(int lottoTicketCount) {
+        return MathUtils.calculateYield(getTotalPrice(), lottoTicketCount * LOTTO_PRICE);
+    }
+
+    private int getTotalPrice() {
+        return statistics.entrySet().stream()
+                .filter(entry -> entry.getValue() >= 1)
+                .map(entry -> entry.getKey().getWinningMoney())
+                .reduce(0, Integer::sum);
```

Comment:
entry 기준으로 stream 코드를 작성하셨군요!

key 기준으로 Stream 코드를 작성해봤으니 참고해주세요:)
```suggestion
        return statistics.keySet()
                .stream()
                .map(key -> key.getWinningMoney() * statistics.get(key))
                .reduce(0,Integer::sum);
```